repositories.bzl

```diff
@@ -14,300 +14,65 @@
-        org_threeten_threetenbp()
-    if not omit_org_yaml_snakeyaml:
-        org_yaml_snakeyaml()
-    if not omit_cglib_cglib:
-        cglib_cglib()
-    if not omit_javax_inject_javax_inject:
-        javax_inject_javax_inject()
-    if not omit_org_ow2_asm_asm:
-        org_ow2_asm_asm()
-
-def com_google_api_codegen_test_repositories(
-        omit_junit_junit = False,
-        omit_com_google_truth_truth = False,
-        omit_com_google_truth_extensions_truth_java8_extension = False,
-        omit_com_google_api_api_compiler_testing = False,
-        omit_pl_pragmatists_JUnitParams = False,
-        omit_org_mockito_mockito_core = False,
-        omit_org_hamcrest_hamcrest_core = False,
-        omit_net_bytebuddy_byte_buddy = False,
-        omit_org_objenesis_objenesis = False,
-        omit_google_java_format_all_deps = False):
-    if not omit_junit_junit:
-        junit_junit()
-    if not omit_com_google_truth_truth:
-        com_google_truth_truth()
-    if not omit_com_google_truth_extensions_truth_java8_extension:
-        com_google_truth_extensions_truth_java8_extension()
-    if not omit_com_google_api_api_compiler_testing:
-        com_google_api_api_compiler_testing()
-    if not omit_pl_pragmatists_JUnitParams:
-        pl_pragmatists_JUnitParams()
-    if not omit_org_mockito_mockito_core:
-        org_mockito_mockito_core()
-    if not omit_org_hamcrest_hamcrest_core:
-        org_hamcrest_hamcrest_core()
-    if not omit_net_bytebuddy_byte_buddy:
-        net_bytebuddy_byte_buddy()
-    if not omit_org_objenesis_objenesis:
-        org_objenesis_objenesis()
-
-def com_google_api_codegen_tools_repositories(omit_google_java_format_all_deps = False):
-    if not omit_google_java_format_all_deps:
-        google_java_format_all_deps()
-
-def com_google_protobuf():
-    # TODO: try loading precompiled protoc (it comes with common protos) and use it for everything.
-    # Using protobuf requires building it before usage, which takes time.
-    git_repository(
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@com_google_api_codegen_properties//:dependencies.properties.bzl", "PROPERTIES")
+
+def com_google_api_codegen_repositories():
+    # Import dependencies shared between Gradle and Bazel (i.e. maven dependencies)
+    for name, artifact in PROPERTIES.items():
+        _maybe(
+            native.maven_jar,
+            name = name,
+            strip_repo_prefix = "maven.",
+            artifact = _fix_bazel_artifact_format(artifact),
+        )
+
+    # Import Bazel-only dependencies (Gradle version will import maven artifacts of same
+    # version, while Bazel will depend on Bazel workspaces). The versions are shared in the
+    # properties file.
+
+    # A temporary hack, to fix incompatibility of protobuf 3.6.1 with new Bazel versions.
+    # We have to do it, because version 3.6.1.3 exists only as a Bazel workspace, but not as
+    # a maven artifact.
+    # Remove once upgraded to 3.7.0
```

Comment:
3.7.0 is out - can we do this now?

Reply:
Maybe, but this would mean diverging from what artman is doing. Also piloting gprc 3.7.0 on all googleapis. I think the chance of catching something is above 50%, I really do not want to mix this work with grpc dependency upgrade.